src/frontend/src/helpers/Storage/Interface.ts

```diff
@@ -0,0 +1,80 @@
+export {AbstractStorage,
+        File, FileID, FileBrief,
+        Project, ProjectID, ProjectBrief,
+        Settings, defaultSettings}
+
+abstract class AbstractStorage {
+  // connect to the database
+  public abstract async connect(): Promise<void>;
+  // projects
+  public abstract async newProject(name: string): Promise<void>;
+  public abstract async getProject(proj: ProjectID): Promise<Project>;
+  public abstract async getProjects(): Promise<ProjectBrief[]>;
+  public abstract async deleteProject(proj: ProjectID): Promise<void>;
+  public abstract async getFiles(proj: ProjectID): Promise<FileBrief[]>;
+  // files
+  public abstract async newFile(proj: ProjectID, filename: string, contents?: string): Promise<FileID>;
+  public abstract async readFile(file: FileID): Promise<File>;
+  public abstract async writeFile(file: FileID, contents: string): Promise<void>;
+  public abstract async setRunFile(file: FileID, question: string): Promise<void>;
+  public abstract async getRunFile(proj: ProjectID, question: string): Promise<FileID|undefined>;
+  public abstract async renameFile(file: FileID, newName: string): Promise<void>;
+  public abstract async deleteFile(file: FileID): Promise<void>;
+  // settings
+  public abstract async setSettings(settings: Settings): Promise<void>;
+  public abstract async getSettings(): Promise<Settings>;
+}
+
+
+type FileID = [string, string]; // compound key
+type ProjectID = string; // alias of name for now
+
+interface File extends FileBrief {
+  id: FileID;
+  name: string; // a file name is (test|q*|common)/name
+  project: ProjectID;
+  contents: string;
+  checksum: string;
+  lastModified: number;
```

Comment:
Shall we just leave it like this? I prefer the naming convention to be consistent.
For the frontend we probably want to commit to a different indexeddb since the current one students use are not working anyways unfortunately. For the backend I think this is the only property that's different. I'm just converting it to snake case when sending message.